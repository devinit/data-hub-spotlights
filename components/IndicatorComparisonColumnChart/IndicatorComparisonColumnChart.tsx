import React, { FunctionComponent } from 'react';
import { EChartsBaseChart } from '../EChartsBaseChart';
import { toBasicAxisData } from '../EChartsBaseChart/utils';
import { EChartOption } from 'echarts';
import { formatNumber, addPrefixAndSuffix, ValueOptions } from '../../utils';
import { formatSeries } from '../ComparisonChartDataHandler/utils';

interface ComponentProps {
  series?: {
    names: [string, string];
    data: [number[], number[]];
  };
  height?: string;
  valueOptions: ValueOptions[];
}

const IndicatorComparisonColumnChart: FunctionComponent<ComponentProps> = ({ valueOptions, ...props }) => {
  if (!props.series) {
    return <div>No Data</div>;
  }

  const options: EChartOption = {
    tooltip: {
      formatter: (params: EChartOption.Tooltip.Format): string => {
        const { seriesName, seriesIndex, value } = params;

        return seriesIndex === 1
          ? formatSeries('', seriesName, value as number, valueOptions[1])
          : formatSeries('', seriesName, value as number, valueOptions[0]);
      }
    },
    legend: { show: false },
    xAxis: {
      data: toBasicAxisData(['Indicators']),
      axisLabel: { show: false }
    },
    yAxis: [
      {
        name: props.series.names[0],
        nameLocation: 'center',
        nameTextStyle: { padding: 30 },
        type: 'value',
        position: 'left',
        axisLabel: {
          formatter: (value: number): string => addPrefixAndSuffix(formatNumber(value, 0), valueOptions[0])
        }
      },
      {
        name: props.series.names[1],
        type: 'value',
        position: 'right',
        nameLocation: 'center',
        nameTextStyle: { padding: 30 },
        axisLabel: {
          formatter: (value: number): string => addPrefixAndSuffix(formatNumber(value, 0), valueOptions[1])
        }
      }
    ],
    color: ['#0089cc', '#eb642b'], // TODO: perhaps configure these in CMS
    series: [
      {
        name: props.series.names[0],
        type: 'bar',
        barWidth: 70,
        barGap: '100%',
        data: toBasicAxisData(props.series.data[0])
      },
      {
        name: props.series.names[1],
        type: 'bar',
        barWidth: 70,
        barGap: '100%',
        yAxisIndex: 1,
        data: toBasicAxisData(props.series.data[1])
      }
    ] as EChartOption.SeriesBar[]
  };

  return <EChartsBaseChart options={options} height={props.height} />;
};

export { IndicatorComparisonColumnChart };
