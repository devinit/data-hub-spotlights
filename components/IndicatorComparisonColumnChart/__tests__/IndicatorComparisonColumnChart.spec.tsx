/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { IndicatorComparisonColumnChart } from '../IndicatorComparisonColumnChart';

describe('IndicatorComparisonColumnChart', () => {
  test('renders default state correctly', () => {
    const renderer = TestRenderer.create(<IndicatorComparisonColumnChart />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
