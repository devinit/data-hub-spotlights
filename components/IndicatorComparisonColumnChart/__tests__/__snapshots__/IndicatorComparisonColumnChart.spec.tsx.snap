// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorComparisonColumnChart renders default state correctly 1`] = `
<div
  style={
    Object {
      "height": "400px",
      "width": "100%",
    }
  }
/>
`;
